qiita_db/metadata_template/base_metadata_template.py

```diff
@@ -772,6 +773,123 @@ def _add_common_creation_steps_to_queue(cls, md_template, obj_id,
                                       ', '.join(["%s"] * len(headers))),
             values, many=True)
 
+    def _add_common_extend_steps_to_queue(self, md_template, conn_handler,
+                                          queue_name):
+        r"""Adds the common extend steps to the queue in conn_handler
+
+        Parameters
+        ----------
+        md_template : DataFrame
+            The metadata template file contents indexed by sample ids
+        conn_handler : SQLConnectionHandler
+            The connection handler object connected to the DB
+        queue_name : str
+            The queue where the SQL statements will be added
+
+        Raises
+        ------
+        QiitaDBError
+            If no new samples or new columns are present in `md_template`
+        """
+        # Check if we are adding new samples
+        sample_ids = md_template.index.tolist()
+        sql = """SELECT sample_id FROM qiita.{0}
+                 WHERE {1}=%s""".format(self._table, self._id_column)
+        curr_samples = set(
```

Comment:
I replaced this line with the call to `keys`. However this is good to know for future PR!